libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/ModelOverview.tsx

```diff
@@ -131,34 +134,109 @@ export class ModelOverview extends React.Component<
-    });
+    this.setState(
+      {
+        selectedDatasetCohorts: this.context.errorCohorts.map((errorCohort) => {
+          return errorCohort.cohort.getCohortID();
+        }),
+        selectedMetrics: defaultSelectedMetrics
+      },
+      () =>
+        this.state.datasetCohortChartIsVisible
+          ? this.updateDatasetCohortStats()
+          : this.updateFeatureCohortStats()
+    );
+  }
+
+  private ifCohortIndexesEquals(a: number[], b: number[]): boolean {
+    return (
+      Array.isArray(a) &&
+      Array.isArray(b) &&
+      a.length === b.length &&
+      a.every((val, index) => val === b[index])
+    );
   }
 
   public componentDidUpdate(): void {
-    const maxCohortId = this.getMaxCohortId();
-    if (maxCohortId > this.state.maxCohortId) {
-      // A cohort has a higher ID than the previously recorded
-      // maximum which indicates that new cohorts were created.
-      const newCohorts = this.context.errorCohorts
-        .filter(
-          (errorCohort) =>
-            errorCohort.cohort.getCohortID() > this.state.maxCohortId
-        )
-        .map((errorCohort) => errorCohort.cohort.getCohortID());
-      this.setState({
-        maxCohortId,
-        selectedDatasetCohorts:
-          this.state.selectedDatasetCohorts?.concat(newCohorts)
-      });
+    const newDatasetCohortIDs = this.context.errorCohorts.map((errorCohort) => {
+      return errorCohort.cohort.getCohortID();
+    });
+    const oldDatasetCohortIDs = this.state.datasetBasedCohorts.map(
+      (errorCohort) => {
+        return errorCohort.cohort.getCohortID();
+      }
+    );
+    console.log(newDatasetCohortIDs);
+    console.log(oldDatasetCohortIDs);
+    if (!this.ifCohortIndexesEquals(newDatasetCohortIDs, oldDatasetCohortIDs)) {
+      let newOldDifference = newDatasetCohortIDs.filter(
+        (x) => !oldDatasetCohortIDs.includes(x)
+      );
+      let oldNewDifference = oldDatasetCohortIDs.filter(
+        (x) => !newDatasetCohortIDs.includes(x)
+      );
+      if (newOldDifference.length > 0) {
+        this.setState(
+          {
+            selectedDatasetCohorts:
+              this.state.selectedDatasetCohorts?.concat(newOldDifference)
+          },
+          () => this.updateDatasetCohortStats()
+        );
+      } else if (oldNewDifference.length > 0) {
+        this.setState(
+          {
+            // Remove the oldNewDifference from selectedDatasetCohorts
+            selectedDatasetCohorts:
+              this.state.selectedDatasetCohorts?.concat(oldNewDifference)
+          },
+          () => this.updateDatasetCohortStats()
+        );
+      }
     }
   }
 
+  private updateDatasetCohortStats(): void {
+    const datasetCohortMetricStats = generateMetrics(
+      this.context.jointDataset,
+      this.context.errorCohorts.map((errorCohort) =>
+        errorCohort.cohort.unwrap(JointDataset.IndexLabel)
+      ),
+      this.context.modelMetadata.modelType
+    );
+    console.log(datasetCohortMetricStats);
```

Comment:
there are several print statements in the PR

Reply:
done. removed.